### PR TITLE
Refactor emails.php to classes

### PR DESCRIPTION
**Pull request by Spudley**
Hi.

I have refactored the email.php file into classes. I hope you like what I've done.

Notes:

- I've stubbed out the existing functions in emails.php so that all they do is call the new code, but I've left them there as removing them would break any plugins that do function overrides on them.

- The new code is largely the same as the existing code, with some tidying up done. Obviously re-arranged a bit to suit the new class structure, but other than that I didn't change much.

- I can't see that the notification suspend functionality is actually used anywhere in the system, but I have nevertheless ported it across.

- I've replaced calls to qa_email_validate() with  PHPMailer::validateAddress() as this is a more robust method. It's arguable whether it's even necessary to check this before trying to send as it's one of the first things that PHPMailer does internally anyway.

### CODE REVIEW

**Comment by svivian** [comment]
Thanks for this! I will review when I can.

File: qa-include/Q2A/Notifications/Email.php [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	 * @param type $handle
+	 * @return \self
+	 * @throws FatalErrorException
+	 */
+	public static function CreateByUserID($userid, $email = '', $handle = '')
+	{
+		if (!(int)$userid) {
+			throw new FatalErrorException('User ID not specified in Notifications/Email/CreateByUserID.');
+		}
+		$needemail = !PHPMailer::validateAddress($email); // take from user if invalid
```
Comment (Mariusz08) [inline qa-include/Q2A/Notifications/Email.php]: PSR-2

File: qa-include/Q2A/Notifications/Email.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	 * Factory method.
+	 * @param type $userid
+	 * @param type $email
+	 * @param type $handle
+	 * @return \self
+	 * @throws FatalErrorException
+	 */
+	public static function CreateByUserID($userid, $email = '', $handle = '')
+	{
+		if (!(int)$userid) {
```
Comment (Mariusz08) [inline qa-include/Q2A/Notifications/Email.php]: PSR-2

File: qa-include/Q2A/Notifications/Mailer.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+		if (qa_opt('smtp_active')) {
+			$this->isSMTP();
+			$this->Host = qa_opt('smtp_address');
+			$this->Port = qa_opt('smtp_port');
+
+			if (qa_opt('smtp_secure')) {
+				$this->SMTPSecure = qa_opt('smtp_secure');
+			} else {
+				$this->SMTPOptions = array(
```
Comment (Mariusz08) [inline qa-include/Q2A/Notifications/Mailer.php]: Short array - `[]`

**Comment by svivian** [comment]
Hi @Spudley, apologies for the massive delay in getting to this. Finally found some time to test and merge. Thanks for the contribution!

FYI I went ahead and converted the classes to use our new namespace system (in the `qa-src/` directory).

**Pull request merged**